### PR TITLE
test: add unit tests for sms sending

### LINKED ISSUE #14726: Tests to verify SMS locking logic
### Is your proposal related to a problem?

https://github.com/calcom/cal.com/pull/14648#issuecomment-2073854579

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds units tests for SMS sending when an event type with an active SMS workflow is booked. Also tests that SMS are not sent if user or team is locked for SMS sending. 

Run `yarn test workflow-notifications.test.ts` 

Fixes #14726

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking/test/workflow-notifications.test.ts [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+        await handleNewBooking(req);
+
+        expectSMSWorkflowToBeTriggered({
+          sms,
+          toNumber: "000",
+        });
+
+        expectWorkflowToBeTriggered({
+          // emailsToReceive: [organizer.email].concat(otherTeamMembers.map(member => member.email)),
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleNewBooking/test/workflow-notifications.test.ts:393]: We currently only send to one of the hosts for team events. This needs to be fixed. I created an issue for that: https://github.com/calcom/cal.com/issues/14739

**Review by joeauyeung — DISMISSED** [review]
Heads up I'm seeing two errors popping up for tests
image
image

**Review by Udit-takkar — COMMENTED** [review]
LGTM. worked fine for me

Screenshot 2024-04-26 at 9 04 50 PM

**Comment by CarinaWolli** [comment]
@joeauyeung did you run migrations?

**Review by hariombalhara — APPROVED** [review]
Nice work @CarinaWolli !!

**Pull request merged**